### PR TITLE
HIPCMS-455

### PR DESCRIPTION
**Pull request by sstamm**

### CODE REVIEW

File: src/Api/Controllers/UsersController.Picture.cs [hunk 1]
```diff
@@ -70,44 +47,23 @@ private IActionResult GetPicture([FromRoute]int userId)
...
-        [HttpPut("{id}/picture/")]
+        [HttpPut("Picture/")]
         [ProducesResponseType(typeof(void), 200)]
         [ProducesResponseType(typeof(void), 400)]
         [ProducesResponseType(typeof(void), 403)]
-        public IActionResult PutPicture([FromRoute]int id, [FromForm] IFormFile file)
+        public IActionResult PutPicture([FromQuery] string identy, [FromForm] IFormFile file)
         {
-            if (!_userPermissions.IsAllowedToAdminister(User.Identity.GetUserId()))
+            if (identy != null && !_userPermissions.IsAllowedToAdminister(User.Identity.GetUserIdenty()))
```
Comment (janis-kra) [inline src/Api/Controllers/UsersController.Picture.cs]: same here, should be `identity`

File: src/Api/Controllers/UsersController.Picture.cs [hunk 2]
```diff
@@ -151,46 +107,25 @@ private static bool IsImage(IFormFile file)
...
-        /// Delete picture for the user {id}
+        /// Delete picture for the current user
         /// </summary>        
-        /// <param name="id">The Id of the user</param>                                 
+        /// <param name="identy">The identy of the user to be delete (For admins)</param>                                 
         /// <response code="200">Request is accepted</response>        
         /// <response code="400">Request incorrect</response>        
         /// <response code="403">User not allowed to delete picture</response>                
         /// <response code="401">User is denied</response>
-        [HttpDelete("{id}/picture/")]
```
Comment (janis-kra) [inline src/Api/Controllers/UsersController.Picture.cs]: so it won't be possible anymore to delete another user's picture -- is that intended?

File: src/Api/Controllers/UsersController.cs [hunk 3]
```diff
@@ -35,7 +36,7 @@ public UsersController(CmsDbContext dbContext, ILoggerFactory logger) : base(dbC
         /// <response code="409">Resource already exists</response>        
         /// <response code="503">Service unavailable</response>        
         /// <response code="401">User is denied</response>
-        [HttpPost("Invite")]
+        [HttpPost("/Api/Users/Invite")]
```
Comment (janis-kra) [inline src/Api/Controllers/UsersController.cs]: This style of setting the URL is inconsistent to the other controllers... shouldn't we create a separate UsersController that has the Invite and GetUsers endpoints, as these are semantically different?

File: src/Api/Controllers/UsersController.cs [hunk 4]
```diff
@@ -75,7 +76,7 @@ public IActionResult InviteUsers([FromBody]InviteFormModel model, [FromServices]
         /// <param name="pageSize">Size of the requested page</param>
         /// <response code="200">Returns PagedResults of UserResults</response>        
         /// <response code="401">User is denied</response>
-        [HttpGet]
+        [HttpGet("/Api/Users/")]
```
Comment (janis-kra) [inline src/Api/Controllers/UsersController.cs]: see above

File: src/Api/Controllers/UsersController.cs [hunk 5]
```diff
@@ -108,88 +109,49 @@ public IActionResult Get([FromRoute]int id)
...
-        [ProducesResponseType(typeof(void), 200)]
-        [ProducesResponseType(typeof(void), 400)]
-        [ProducesResponseType(typeof(void), 404)]
-        public IActionResult Put([FromBody]UserFormModel model)
-        {
-            return PutUser(User.Identity.GetUserId(), model);
-        }
-
-        // PUT api/users/:id
+        // PUT api/users
```
Comment (janis-kra) [inline src/Api/Controllers/UsersController.cs]: should be `api/user` without the s

File: src/Api/Managers/AttachmentsManager.cs [hunk 6]
```diff
@@ -27,22 +27,17 @@ public IEnumerable<TopicAttachmentResult> GetAttachments(int topicId)
             return DbContext.TopicAttatchments.Where(ta => (ta.TopicId == topicId)).Include(ta => ta.User).Include(ta => ta.Legal).ToList().Select(at => new TopicAttachmentResult(at));
         }
 
-        public EntityResult CreateAttachment(int topicId, int userId, AttatchmentFormModel model)
+        public EntityResult CreateAttachment(int topicId, string userIdenty, AttatchmentFormModel model)
```
Comment (janis-kra) [inline src/Api/Managers/AttachmentsManager.cs]: parameter also named wrong

File: src/Api/Managers/Notification/NotificationManager.cs [hunk 7]
```diff
@@ -13,8 +13,9 @@ public class NotificationManager : BaseManager
     {
         public NotificationManager(CmsDbContext dbContext) : base(dbContext) { }
 
-        public IEnumerable<NotificationResult> GetNotificationsForTheUser(int userId, bool onlyUreard)
+        public IEnumerable<NotificationResult> GetNotificationsForTheUser(string userIdenty, bool onlyUreard)
```
Comment (janis-kra) [inline src/Api/Managers/Notification/NotificationManager.cs]: `onlyUnread`

File: src/Api/Models/User/UserResult.cs [hunk 8]
```diff
@@ -4,7 +4,7 @@ public class UserResult
     {
         public UserResult(Entity.User user)
         {
-            Id = user.Id;
+            Identy = user.Email; // TODO replace with 'identy'
```
Comment (janis-kra) [inline src/Api/Models/User/UserResult.cs]: When is that going to happen?

Comment (sstamm) [inline src/Api/Models/User/UserResult.cs]: When we have a 'real' identity xD Eg. when swichting to IdentityServer.

File: src/Api/Controllers/UsersController.Picture.cs (hunk 2 above)
Comment (sstamm) [inline src/Api/Controllers/UsersController.Picture.cs]: No:
        public IActionResult Delete([FromQuery] string identity)
        {
            if (identity == null && !_userPermissions.IsAllowedToAdminister(User.Identity.GetUserIdentity()))
                return Forbidden();

"[FromQuery] string identity" is an optinal Parameter, and only Admins can specify the identity.

**Pull request merged**